crates/iceberg-catalog/src/catalog/io.rs

```diff
@@ -19,6 +115,9 @@ pub(crate) async fn write_metadata_file(
     };
     tracing::debug!("Going to write metadata file to {}", metadata_location);
 
+    let compression_codec =
+        CompressionCodec::try_from_metadata(&metadata).map_err(IoError::UnknownCompressionCodec)?;
+
```

Comment:
Currently, it would fail earlier, when the location for metadata is calculated using `fn initial_metadata_location`, which also needs a `CompressionCodec`. The compiler can't know this, nor can we be sure this will always be the case. I will change it to `compression_codec` beeing a parameter.